test/jdk/java/net/DatagramSocket/SendDatagramToBadAddress.java

```diff
@@ -170,6 +175,8 @@ private void test(DatagramSocket sock) throws Exception {
                 sock.send(p);
                 p = new DatagramPacket(buf, buf.length, addr, port);
                 sock.receive (p);
+                print("(unexpectedly) received data from address " + p.getAddress()
+                        + " port " + p.getPort() + " on attempt " + i);
             } catch (InterruptedIOException ex) {
                 print ("socket timeout");
             } catch (Exception ex) {
```

Comment:
Can we catch `PortUnreachableException` here instead?

Reply:
Here too I didn't want to change the current behaviour/code of the test. It's not just this catch block but even the one a few lines above which catches `InterruptedIOException`. Neither the `send()` nor the `receive()` APIs of `DatagramSocket` specify that they throw this specific exception, but the test seems to catch it and consider it a socket timeout. So I'm guessing this test and the catch block was written with some specific context in mind and I didn't want to change that part as part of this PR which only aims to enable logging by default.

Reply:
Right. Looks like `OSsupportsFeature` was used to filter out systems that don't send "port unreachable"; we could probably use it here if the failure turns out to be OS-specific.

Additional logs look fine.

Reply:
as I indicated below, we have tracked and investigated this issue and it is not purely a macosx-aarch64 issue. It may also be a  test env issue. As such using the OS filtering in the test, which is designed to primarily exclued AIX, wouldn't be the appropraite approach. If we wish to exclude the test on macosx-aarch64, then use of Problemlist is more appropriate.

Reply:
Hello Mark,

I just noticed this comment. My understanding of the comment is that you don't expect any other changes to this test other than what has already been done in this PR to enable the debug logs? Is that correct?

Reply:
yes, the addition of the diagnostics is useful